### PR TITLE
chore: update golang workflow and support ci-lint for golang

### PR DESCRIPTION
**Pull request by Chever-John**
link to #32

### CODE REVIEW

**Comment by Softwarekang** [comment]
Ignore example directory and fix other golangci-lint err
@Chever-John :smile:

**Comment by Chever-John** [comment]
@Softwarekang Hi, maybe you should see this commit([f813f11](https://github.com/Softwarekang/knetty/pull/34/commits/f813f118516b79ca0313750df24016773a99363f))

File: client.go [hunk 1]
```diff
@@ -18,6 +18,7 @@ type Client struct {
 }
 
 // NewClient init the client
+//
```
[the remark this answers is not in the document]
Comment (Softwarekang) [reply client.go:21]: 同样规范一下 import分层

**Review by Softwarekang — CHANGES_REQUESTED** [review]

File: session/session.go [hunk 2]
```diff
@@ -4,10 +4,9 @@ package session
 import (
 	"errors"
 	"fmt"
-	"log"
-
 	"github.com/Softwarekang/knetty/net/connection"
 	"github.com/Softwarekang/knetty/pkg/buffer"
+	"log"
```
[the remark this answers is not in the document]
Comment (Chever-John) [reply session/session.go:9]: maybe we can add gofmt into makefile

File: net/poll/poller_manager.go [hunk 3]
```diff
@@ -62,7 +62,12 @@ func (m *pollerManager) Run() error {
 	for idx := len(m.pollers); idx < m.NumLoops; idx++ {
 		var poller = NewDefaultPoller()
 		m.pollers = append(m.pollers, poller)
-		go poller.Wait()
+		go func() {
+			err := poller.Wait()
```
[the remark this answers is not in the document]
Comment (Chever-John) [reply net/poll/poller_manager.go]: Solved

**Review by Softwarekang — APPROVED** [review]
LGTM

**Pull request merged**